### PR TITLE
34g842d: refactoring of the GunPartCard component

### PR DESCRIPTION
**Pull request by Logovas1994**

### CODE REVIEW

**Comment by AbrosimovVladislav** [comment]
Brands start jumping if we have longer names
Снимок экрана 2022-12-08 в 18 53 56

**Comment by nastyakuzusha** [comment]
Screenshot 2022-12-09 at 18 40 07
cannot build this page

File: front/src/pages/nft/[id].tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ const SingleNFTPage = (props: SingleNFTPageProps) => {
                     </div>
                     <NftCardInformation data={data}></NftCardInformation>
```
Comment (nastyakuzusha) [inline front/src/pages/nft/[id].tsx]: you should handle situations when server returns 404 here, for now you could basically set up data?... for every place you are trying to map it

Comment (nastyakuzusha) [reply front/src/pages/nft/[id].tsx]: and remove please closing tag here, even IDEA shows warning here
<NFTCardInformation .../>

**Review by AbrosimovVladislav — APPROVED** [review]

File: front/src/pages/nft/[id].tsx [hunk 2]
```diff
@@ -32,15 +32,15 @@ const SingleNFTPage = (props: SingleNFTPageProps) => {
                     <div className={classes.nftWrapper}>
                         <GCImage className={classes.nftImage} src={data.nftImageUrl} alt="gun"/>
                     </div>
-                    <NftCardInformation data={data}></NftCardInformation>
+                    <NftCardInformation data={data}/>
```
Comment (nastyakuzusha) [inline front/src/pages/nft/[id].tsx:35]: you need also to check data for null inside NFTCardInformation component

File: front/src/pages/nft/[id].tsx [hunk 3]
```diff
@@ -32,15 +32,15 @@ const SingleNFTPage = (props: SingleNFTPageProps) => {
...
                 </div>
                 <GCGrid>
                     <GCGridCol sm={12} md={12}>
                         <GCText className={classes.catalogHeader} h2 bold>
                             What was used in this build
                         </GCText>
                         <Catalog>
-                            {data.properties.map((product: Product) => (
+                            {data?.properties.map((product: Product) => (
```
Comment (nastyakuzusha) [inline front/src/pages/nft/[id].tsx]: properties can also be null even if data exists

Comment (nastyakuzusha) [reply front/src/pages/nft/[id].tsx]: please, add ? to properties

**Pull request merged**